paging/common/src/main/kotlin/androidx/paging/InvalidatingPagingSourceFactory.kt

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2020 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.paging
+
+import androidx.annotation.VisibleForTesting
+
+/**
+ * A PagingSource Factory that can be kept as a reference and stores a list of paging sources
+ * created with the factory. Includes invalidate() function to invalidate the list of [PagingSource]
+ * when data changes.
```

Comment:
```suggestion
 * Abstract [PagingSource] factory intended for usage in [Pager] construction, that tracks [PagingSource]s
 * it produces for invalidation when the backing dataset is updated.
 *
 * Calling [invalidate] on this [InvalidatingPagingSourceFactory] will automatically forward 
 * invalidate signals to all active [PagingSource]s.
```

Reply:
Using [ClassName] or [function] will let KDocs link to the class, which is super helpful as it shows up as a clickable link in the generated docs.